### ISSUE TITLE
Cannot install on Windows 10 due to incompatibility with python 3.10

### ISSUE THREAD

**Issue by animeshkrishnabulusu** [lektor/lektor#943]
I am unable to install lektor on a new Windows 10 with `python 3.10.0`

I tried both the official way as well as `pipx` with no luck. I get the same error.

Any idea on what could be wrong or what I could be doing wrong here?

```
PS C:\Users\animesh> pipx install lektor
  installed package lektor 3.2.2, Python 3.10.0
  These apps are now globally available
    - lektor.exe
done! ✨ 🌟 ✨
PS C:\Users\animesh> lektor
Traceback (most recent call last):
  File "C:\Users\animesh\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\animesh\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\animesh\.local\bin\lektor.exe\__main__.py", line 4, in <module>
  File "C:\Users\animesh\.local\pipx\venvs\lektor\lib\site-packages\lektor\cli.py", line 13, in <module>
    from .project import Project
  File "C:\Users\animesh\.local\pipx\venvs\lektor\lib\site-packages\lektor\project.py", line 5, in <module>
    from inifile import IniFile
  File "C:\Users\animesh\.local\pipx\venvs\lektor\lib\site-packages\inifile.py", line 7, in <module>
    from collections import MutableMapping, OrderedDict
ImportError: cannot import name 'MutableMapping' from 'collections' (C:\Users\animesh\AppData\Local\Programs\Python\Python310\lib\collections\__init__.py)
PS C:\Users\animesh>
```

**Comment by andoresuperesu** [comment]
This happens because the collections abstract base classes were moved to collections.abc submodule. That would require a change on inifile package (https://github.com/mitsuhiko/python-inifile) or to have its corrected code embedded into lektor (unlikely).

A temporary quickfix would require the following:

Open the inifile.py located in `C:\Users\animesh\.local\pipx\venvs\lektor\lib\site-packages\inifile.py` and modify line 7.
```python
from collections import MutableMapping, OrderedDict
```

to the following:

```python
from collections.abc import MutableMapping
from collections import OrderedDict
```

A permanent solution would require a pr to https://github.com/mitsuhiko/python-inifile adressing issue https://github.com/mitsuhiko/python-inifile/issues/11.

**Comment by dairiki** [comment]
To be clear, this is not a Windows-specific issue.  It affects lektor run under python 3.10 on any platform.

> A permanent solution would require a pr to https://github.com/mitsuhiko/python-inifile adressing issue [mitsuhiko/python-inifile#11](https://github.com/mitsuhiko/python-inifile/issues/11).

Such a PR (mitsuhiko/python-inifile#10) has existed since February, with no comments from any maintainers. :-(
(Python-inifile appears to have not seen much love for quite awhile.)

The easiest short-term work-around is probably to run lektor using python < 3.10 (e.g. using python 3.9) if possible.

Longer term, if no fix in python-inifile is forthcoming, lektor is either going to have to:
- Vendor-branch (or fork) python-inifile, so that we can fix it ourself.  (yuck)
- Use some other library for inifile parsing (maybe [configparser](https://docs.python.org/3/library/configparser.html) from the python stdlib.  This would require some code changes, but I think would not be a huge project.

**Comment by animeshkrishnabulusu** [comment]
I changed the inifile imports as above and `lektor` command works. Now when I do `lektor build`, it complains about 
```
Installing collected packages: Werkzeug, MarkupSafe, lektor-atom
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
flask 2.0.2 requires Werkzeug>=2.0, but you have werkzeug 0.16.1 which is incompatible.
Successfully installed MarkupSafe-2.0.1 Werkzeug-0.16.1 lektor-atom-0.3.1
```

[Full error here](https://gist.github.com/anmsh/2b83f32a31ab7aa31bf033b9bb04a183)

The same thing happens when I drop down to `python 3.9`

**Comment by andoresuperesu** [comment]
@anmsh

Yes, that's unrelated to this issue.

**Comment by animeshkrishnabulusu** [comment]
Agreed and I think this is the issue PR related to this werkzeug error.

#911

**Comment by andoresuperesu** [comment]
Nope, it was solved and it was included in the latest release of lektor.

**Comment by animeshkrishnabulusu** [comment]
I see that in the releases that this fix went in. However, I get this even now.
Noticed that my lektor-atom plugin was  at version 0.3.1 which had the requirement of werkzeug 1x. I updated to version 0.4 and that fixed it.

Thanks for all the help.

**Comment by andoresuperesu** [comment]
@anmsh Could you keep the issue open and rephrase it to mention only the python 3.10 compatibility issue?

**Comment by animeshkrishnabulusu** [comment]
@andresperezcera I got a bit too excited there.
Thanks, I have reworded the title a bit.

**Comment by dairiki** [comment]
This is fixed in #938 (and #953).
